File: server/src/main/java/it/podlodowski/aeronmgmt/server/events/EventSeverity.java
package it.podlodowski.aeronmgmt.server.events;

import java.util.Map;

public enum EventSeverity {
    ERROR, WARNING, INFO, SUCCESS;

    private static final Map<String, EventSeverity> SEVERITY_MAP = Map.ofEntries(
            // Error — needs attention
            Map.entry("CONSENSUS_LOST", ERROR),
            Map.entry("NODE_DOWN", ERROR),
            Map.entry("NODE_CNC_LOST", ERROR),
            Map.entry("AGENT_DISCONNECTED", ERROR),
            Map.entry("MONITORING_GAP", ERROR),

            // Warning — notable
            Map.entry("ELECTION_STARTED", WARNING),
            Map.entry("CLUSTER_SUSPENDED", WARNING),
            Map.entry("CLUSTER_SHUTDOWN", WARNING),

            // Success — positive
            Map.entry("CONSENSUS_ESTABLISHED", SUCCESS),
            Map.entry("NODE_UP", SUCCESS),
            Map.entry("AGENT_CONNECTED", SUCCESS),
            Map.entry("NODE_CNC_ACCESSIBLE", SUCCESS),
            Map.entry("CLUSTER_START", SUCCESS),
            Map.entry("CLUSTER_RESUMED", SUCCESS),

            // Info — everything else handled by default
            Map.entry("LEADER_ELECTED", INFO),
            Map.entry("ELECTION_COMPLETED", INFO),
            Map.entry("ROLE_CHANGE", INFO),
            Map.entry("MODULE_STATE_CHANGE", INFO),
            Map.entry("SNAPSHOT_TAKEN", INFO),
            Map.entry("SNAPSHOT_REQUESTED", INFO),
            Map.entry("NODE_ACTION", INFO),
            Map.entry("EGRESS_RECORD_STARTED", INFO),
            Map.entry("EGRESS_RECORD_STOPPED", INFO)
    );

    public static EventSeverity fromType(String eventType) {
        return SEVERITY_MAP.getOrDefault(eventType, INFO);
    }
}

File: ui/src/utils/eventSeverity.ts
export type EventSeverity = 'error' | 'warning' | 'info' | 'success'

const SEVERITY_MAP: Record<string, EventSeverity> = {
  // Error — needs attention
  CONSENSUS_LOST: 'error',
  NODE_DOWN: 'error',
  NODE_CNC_LOST: 'error',
  AGENT_DISCONNECTED: 'error',
  MONITORING_GAP: 'error',

  // Warning — notable
  ELECTION_STARTED: 'warning',
  CLUSTER_SUSPENDED: 'warning',
  CLUSTER_SHUTDOWN: 'warning',

  // Success — positive
  CONSENSUS_ESTABLISHED: 'success',
  NODE_UP: 'success',
  AGENT_CONNECTED: 'success',
  NODE_CNC_ACCESSIBLE: 'success',
  CLUSTER_START: 'success',
  CLUSTER_RESUMED: 'success',

  // Info — everything else
  LEADER_ELECTED: 'info',
  ELECTION_COMPLETED: 'info',
  ROLE_CHANGE: 'info',
  MODULE_STATE_CHANGE: 'info',
  SNAPSHOT_TAKEN: 'info',
  SNAPSHOT_REQUESTED: 'info',
  NODE_ACTION: 'info',
  EGRESS_RECORD_STARTED: 'info',
  EGRESS_RECORD_STOPPED: 'info',
}

export function getEventSeverity(eventType: string): EventSeverity {
  return SEVERITY_MAP[eventType] ?? 'info'
}

export const SEVERITY_BADGE: Record<EventSeverity, string> = {
  error: 'bg-critical-surface text-critical-text',
  warning: 'bg-warning-surface text-warning-text',
  info: 'bg-elevated text-text-secondary',
  success: 'bg-elevated text-text-secondary',
}

export const SEVERITY_FILL: Record<EventSeverity, string> = {
  error: 'var(--critical-text)',
  warning: 'var(--warning-text)',
  info: 'var(--info-text)',
  success: 'var(--success-text)',
}
